scripts/multi_component_hydrogen_bond_propensity/multi_component_hydrogen_bond_propensity_report.py

```diff
@@ -314,14 +313,18 @@ def make_mc_report(identifier, results, directory, diagram_file, chart_file):
     launch_word_processor(output_file)
 
 
-def main(structure, work_directory, failure_directory, library, csdrefcode):
+def main(structure, work_directory, failure_directory, library, csdrefcode, force_run):
     # This loads up the CSD if a refcode is requested, otherwise loads the structural file supplied
     if csdrefcode:
         try:
             crystal = io.CrystalReader('CSD').crystal(structure)
         except RuntimeError:
             print('Error! %s is not in the database!' % structure)
             quit()
+        if io.CrystalReader('CSD').entry(structure).has_disorder and not force_run:
+            raise RuntimeError("Disorder can cause undefined behaviour. It is not advisable to run this "
+                               "script on disordered entries.\n To force this script to run on disordered entries"
+                               " use the flag --force_run_disordered.")
```

Comment:
We are working on better disorder handling for structures already in the CSD so when that project is done this might need to be edited as it will be possible to run on one of the assemblies of the disordered molecule

Reply:
Yes, I looked into it and could not find on the documentation a way to do that. But if it is on the works that's nice. Will it automatically do it? Because if the script will have to be modified anyway after the fact, I don't see the problem in using this as a bandaid. Furthermore, someone could go on Mercury and create a mol2 without disorder for running. So I think this is a good solution for now. As having a warning is better than having weird errors pop up later. In the last case, it is trivial to get the file before this changes from git history.

Reply:
There is already the option to do this for CIFs, with crystal.disordered_molecule, but not refcodes yet. It won't be automatic so it will still need a new edit but I agree that for now this works. Was just letting you know that at some point the handling of disorder will be better :)